aiida_vasp/calcs/workchains/converge.py

```diff
@@ -0,0 +1,1392 @@
+        kstep = self.ctx.converge.k_sampling[self.ctx.converge.kpoints_iteration]
+        rec_cell = self.ctx.converge.kpoints.cell
+        kgrid = fetch_k_grid(rec_cell, kstep)
+        self.ctx.converge.settings['kgrid'] = kgrid
+        # Update grid.
+        kpoints = get_data_class('array.kpoints')
+        self.ctx.converge.kpoints = kpoints(kpoints_mesh=kgrid, cell_from_structure=self.ctx.inputs.structure)
+        self.ctx.running_kpoints = True
+        self.ctx.running_pw = False
+        inform_details = self.ctx.converge.settings.get('inform_details')
+        if inform_details is None:
+            inform_details = ''
+
+        # inform user
+        if self._verbose:
+            self.report('running k-point convergence test for k-point sampling '
+                        'of {}x{}x{} for a plane wave cutoff {encut}'.format(
+                            kgrid[0], kgrid[1], kgrid[2], encut=self.ctx.converge.settings['encut']) + inform_details)
+
+        return
+
+    def results_kpoints_conv_calc(self):
+        """Fetch and store the relevant convergence parameters for each k-point grid calculation."""
+
+        # Check if child workchain was successfull
+        exit_status = self.ctx.kpoints_workchains[-1].exit_status
+        if exit_status:
+            self.report('This single convergence calculation has to be considered '
+                        'failed as the exit status from a child workchain is not '
+                        'zero, exit_status:{}'.format(exit_status))
+        # Update kpoints iteration index
+        self.ctx.converge.kpoints_iteration += 1
+        # Check if the index has an entry, if not, do not perform further calculations
+        try:
+            self.ctx.converge.k_sampling[self.ctx.converge.kpoints_iteration]
+        except IndexError:
+            self.ctx.converge.run_kpoints_conv_calcs = False
+
+        try:
+            self.ctx.kpoints_workchains[-1]
+        except IndexError:
+            self.report('the k-point grid convergence calculation ' 'finished without returning a {}'.format(self._next_workchain.__name__))
+
+        kgrid = self.ctx.converge.settings['kgrid']
+        encut = self.ctx.converge.settings['encut']
+        if not exit_status:
+            # fetch total energy
+            energy = 0.0
+            #energy = workchain.out['output_energies'].get_array('energy_no_entropy')
+
+            # fetch force
+            #forces = workchain.out['output_final_forces'].get_array('forces')
+
+            # locate maximum force
+            #max_force = np.amax(forces)
+            max_force = 0.0
+
+            # fetch bands and occupations
+            #bands = workchain.out['output_bands']
+
+            # find VBM and band gap (currently only do band gap
+            # in order to use internal Aiida function)
+            max_valence_band = 0.0
+            #_, gap = find_bandgap(bands)
+            gap = 0.0
```

Comment:
should these be computed instead of just set to `0.0` some time before merging? It seems incomplete.

Reply:
If so, consider changing the tick box in the PR description to "not ready" until this is done.

Reply:
Or alternatively document it as a missing feature and create an issue.

Reply:
Absolutely. In fact this has not been done as the properties have not yet been added to the parser. Working on that now. Also, we need to calculate the band gap. I have functions for this, so does Aiida, but I have not looked into which is the better approach yet. We should use Aiida stuff, so if that needs improvements, we should improve that as well. Thanks for changing status to wip.